File: AmsterdamEats/AmsterdamEats/Models/SearchCriteria.swift
import Foundation

struct SearchCriteria {
    var date: Date = Date()
    var time: Date = Calendar.current.date(bySettingHour: 19, minute: 0, second: 0, of: Date()) ?? Date()
    var numberOfPeople: Int = 2
    var cuisineType: CuisineType = .any
    var priceRange: Set<PriceLevel> = [.moderate]

    enum CuisineType: String, CaseIterable, Identifiable {
        case any, italian, japanese, french, dutch, indian, thai, american, mediterranean, chinese, mexican
        var id: String { rawValue }
        var displayName: String { rawValue.capitalized }
        var queryPrefix: String { self == .any ? "restaurant" : "\(rawValue) restaurant" }
    }

    enum PriceLevel: String, CaseIterable, Identifiable, Hashable {
        case budget = "PRICE_LEVEL_INEXPENSIVE"
        case moderate = "PRICE_LEVEL_MODERATE"
        case expensive = "PRICE_LEVEL_EXPENSIVE"
        case veryExpensive = "PRICE_LEVEL_VERY_EXPENSIVE"
        var id: String { rawValue }
        var displayName: String {
            switch self {
            case .budget: return "€  (Budget)"
            case .moderate: return "€€  (Moderate)"
            case .expensive: return "€€€  (Expensive)"
            case .veryExpensive: return "€€€€  (Fine Dining)"
            }
        }
    }
}

File: AmsterdamEats.Maui/AmsterdamEats.Maui/Models/SearchCriteria.cs
namespace AmsterdamEats.Models;

public class SearchCriteria
{
    public DateTime Date { get; set; } = DateTime.Today.AddDays(1);
    public TimeSpan Time { get; set; } = new TimeSpan(19, 0, 0);
    public int NumberOfPeople { get; set; } = 2;
    public CuisineType Cuisine { get; set; } = CuisineType.Any;
    public HashSet<PriceLevel> PriceRange { get; set; } = [PriceLevel.Moderate];
    public string DateString => Date.ToString("yyyy-MM-dd");
}

public enum CuisineType
{
    Any, Italian, Japanese, French, Dutch, Indian, Thai, American, Mediterranean, Chinese, Mexican
}

public enum PriceLevel
{
    Budget, Moderate, Expensive, VeryExpensive
}
